_pages/documentation/gem5-stdlib/7-using-local-data-sources.md

```diff
@@ -0,0 +1,168 @@
+
+```python
+resource = obtain_resource("id", clients=["data-source-1"])
+```
+
+### Using Custom Resource Configuration and Local Resources
+
+In this example, we will walk through how to set up your custom configuration and utilize your own local resources. For this illustration, we'll employ a JSON file as our resource data source.
+
+#### Creating a Custom Resource Data Source
+
+Let's begin by downloading the resource we'll use as our local source. I'm selecting the `riscv-hello` resource from gem5 resources.
+
+```bash
+wget https://storage.googleapis.com/dist.gem5.org/dist/develop/test-progs/hello/bin/riscv/linux/hello-20220728
+```
+
+Next, let's create the JSON data source. I'll name the file `my-resources.json`. The contents should look like this:
+
+```json
+[
+    {
+        "category": "binary",
+        "id": "riscv-hello",
+        "description": "A 'Hello World!' binary, compiled to RISCV.",
+        "architecture": "RISCV",
+        "size": 4674040,
+        "tags": [
+            "hello"
+        ],
+        "is_zipped": false,
+        "md5sum": "6d9494d22b90d817e826b0d762fda973",
+        "source": "src/simple",
+        "url": "file://<path/to/the/above/downloaded/file>",
+        "code_examples": [
+            {
+                "example": "https://github.com/gem5/gem5/tree/develop/configs/example/gem5_library/checkpoints/riscv-hello-restore-checkpoint.py",
+                "tested": true
+            },
+            {
+                "example": "https://github.com/gem5/gem5/tree/develop/configs/example/gem5_library/checkpoints/riscv-hello-save-checkpoint.py",
+                "tested": true
+            },
+            {
+                "example": "https://github.com/gem5/gem5/tree/develop/configs/example/gem5_library/riscvmatched-hello.py",
+                "tested": true
+            }
+        ],
+        "license": "",
+        "author": [],
+        "source_url": "https://github.com/gem5/gem5-resources/tree/develop/src/simple",
+        "resource_version": "1.0.0",
+        "gem5_versions": [
+            "23.0"
+        ],
+        "example_usage": "obtain_resource(resource_id=\"riscv-hello\")"
+    }
+]
+```
+
+The JSON object of a resource should adhere to the [gem5 resources schema](https://resources.gem5.org/gem5-resources-schema.json). 
+
+**Note**: While the `url` field can be a link, in this case, I'm using a local download.
+
+#### Creating Your Custom Resource Configuration
+
+Create a file named `gem5-config.json` with the following content:
+
+```json
+{
+    "sources": {
+        "my-json-data-source": {
+            "url": "path/to/my-resources.json",
+            "isMongo": false
```

Comment:
Worth explaining this very briefly: I.e., `(isMongo == True) == (isJSON == False)` :)